include/qfusion.hpp

```diff
@@ -0,0 +1,177 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano and the Qrack contributors 2017, 2018. All rights reserved.
+//
+// QUnit maintains explicit separability of qubits as an optimization on a QEngine.
+// See https://arxiv.org/abs/1710.05867
+// (The makers of Qrack have no affiliation with the authors of that paper.)
+//
+// Licensed under the GNU Lesser General Public License V3.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/lgpl-3.0.en.html
+// for details.
+
+#pragma once
+
+#include "qinterface.hpp"
+
+namespace Qrack {
+
+class QFusion;
+typedef std::shared_ptr<QFusion> QFusionPtr;
+
+class QFusion : public QInterface {
+protected:
+    static const bitLenInt MIN_FUSION_BITS = 3U;
+    QInterfacePtr qReg;
+    QInterfaceEngine engineType;
```

Comment:
This is never used in QFusion.

Reply:
It is used to set the threshold for gate fusion in ApplySingleBit and Decohere.

Reply:
Removed engineType.